test/beef_test.go

```diff
@@ -62,6 +72,32 @@ func TestBeef(t *testing.T) {
 			statusResponse, err := arcClient.GETTransactionStatusWithResponse(context.Background(), tx.TxID())
 			require.NoError(t, err)
 			require.Equal(t, metamorph_api.Status_MINED.String(), *statusResponse.JSON200.TxStatus)
+
+			// verify callbacks for both unmined txs in BEEF
+			lastTxCallbackReceived := false
+			middleTxCallbackReceived := false
+
+			for i := 0; i < 2; i++ {
```

Comment:
Are 2 callbacks expected?
Maybe add a variable there like `expectedCallbacks := 2` to make it more explicit

Reply:
Yes, 2 callbacks are expected, one for the last tx in beef and one for the unmined parent, as per this feature request.

Thanks for pointing that out, magic numbers are the worst. Fixed

Reply:
IMO, making this magic number a variable doesn't make the code more readable or clear; moreover, it is misleading.
1. Making it a variable lets other developers think it is configurable via the variable, which is not true.
2. The expected callback number is hardcoded in the prepareBeef() function.

To make it truly readable, we should:
1. Change the prepareBeef() function implementation to make it possible to configure how many transactions it should contain (mined and unmined) – IMO, overengineering.
2. Return the number of "unmined" transactions from prepareBeef() – this way, we are hiding all the mess from the test code and move it to the setup code.